File: src/main/java/de/dc/planetsudo/level/save/BackToBack.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package de.dc.planetsudo.level.save;

import de.dc.util.data.Base2D;
import de.dc.util.data.Direction2D;
import java.awt.Color;
import java.awt.Polygon;
import de.dc.planetsudo.level.AbstractLevel;
import de.dc.planetsudo.level.ResourcePlacement;
import de.dc.planetsudo.level.ResourceRandomRectangle;
import de.dc.planetsudo.level.levelobjects.Resource.ResourceType;

/**
 *
 * @author divine
 */
public class BackToBack extends AbstractLevel {

	public BackToBack() {
	}

	@Override
	protected Polygon loadLevelBorderPolygon() {
		Polygon levelBorders = new Polygon();
		levelBorders.addPoint(0, 0);
		levelBorders.addPoint(1000, 0);
		levelBorders.addPoint(1000, 1000);
		levelBorders.addPoint(0, 1000);
		return levelBorders;
	}

	@Override
	protected Base2D[] loadHomePositions() {
		Base2D[] homePosition = new Base2D[2];
		homePosition[0] = new Base2D(390, 450, Direction2D.RIGHT);
		homePosition[1] = new Base2D(610, 450, Direction2D.RIGHT);
		return homePosition;
	}

	@Override
	protected ResourcePlacement[] loadResourcePlacement() {
		ResourcePlacement[] resourcePlacement = new ResourcePlacement[5];
		resourcePlacement[0] = new ResourceRandomRectangle(0, 0, 1000, 1000, 10, ResourceType.Normal);
		resourcePlacement[1] = new ResourceRandomRectangle(0, 0, 1000, 1000, 5, ResourceType.DoublePoints);
		resourcePlacement[2] = new ResourceRandomRectangle(400, 800, 200, 200, 10, ResourceType.ExtremPoint);
		resourcePlacement[3] = new ResourceRandomRectangle(0, 0, 1000, 1000, 10, ResourceType.ExtraAgentFuel);
		resourcePlacement[4] = new ResourceRandomRectangle(400, 0, 200, 200, 10, ResourceType.ExtraMothershipFuel);
		return resourcePlacement;
	}

	@Override
	protected Color loadLevelColor() {
		return new Color(  244,164,96  );
	}

	@Override
	protected Polygon[] loadLevelWallPolygons() {
                  Polygon[] walls = new Polygon[1];
                    walls[0] = new Polygon();
                         walls[0].addPoint(450, 350);
                         walls[0].addPoint(450, 550);
                         walls[0].addPoint(550, 550);
                         walls[0].addPoint(550, 350);                                                                               
		return walls;
	}
}

File: src/main/kotlin/org/openbase/planetsudo/level/save/BackToBack.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openbase.planetsudo.level.save

import org.openbase.planetsudo.geometry.Base2D
import org.openbase.planetsudo.geometry.Direction2D
import org.openbase.planetsudo.level.AbstractLevel
import org.openbase.planetsudo.level.ResourceRandomRectangle
import org.openbase.planetsudo.level.levelobjects.Resource.ResourceType
import java.awt.Color
import java.awt.Polygon

/**
 *
 * @author [Divine Threepwood](mailto:divine@openbase.org)
 */
class BackToBack : AbstractLevel() {
    override fun loadLevelBorderPolygon() = Polygon().apply {
        addPoint(0, 0)
        addPoint(1000, 0)
        addPoint(1000, 1000)
        addPoint(0, 1000)
    }

    override fun loadHomePositions() = listOf(
        Base2D(390.0, 450.0, Direction2D.LEFT),
        Base2D(610.0, 450.0, Direction2D.RIGHT),
    )

    override fun loadResourcePlacement() = listOf(
        ResourceRandomRectangle(0, 0, 1000, 1000, 10, ResourceType.Normal),
        ResourceRandomRectangle(0, 0, 1000, 1000, 5, ResourceType.DoublePoints),
        ResourceRandomRectangle(400, 800, 200, 200, 10, ResourceType.ExtremPoint),
        ResourceRandomRectangle(0, 0, 1000, 1000, 10, ResourceType.ExtraAgentFuel),
        ResourceRandomRectangle(400, 0, 200, 200, 10, ResourceType.ExtraMothershipFuel),
        ResourceRandomRectangle(0,0,50,50,5,ResourceType.Tonic),
        ResourceRandomRectangle(0,950,50,50,5,ResourceType.Tonic),
        ResourceRandomRectangle(950,0,50,50,5,ResourceType.Tonic),
        ResourceRandomRectangle(950,950,50,50,5,ResourceType.Tonic)
    )

    override fun loadLevelColor() = Color(244, 164, 96)

    override fun loadLevelWallPolygons() = listOf(
        Polygon().apply {
            addPoint(450, 350)
            addPoint(450, 550)
            addPoint(550, 550)
            addPoint(550, 350)
        },
    )
}
